src/components/StyleGuide/ContainerQuery.mdx

```diff
@@ -0,0 +1,250 @@
+import { Meta, Source } from "@storybook/blocks";
+
+import Link from "../Link/Link";
+
+<Meta title="Development Guide/Container Query Guide" />
+
+# Container Query Guide
+
+## Table of Contents
+
+- {<Link href="#general-information" target="_self">General Information</Link>}
+- {<Link href="#updating-a-component-to-use-container-queries" target="_self">Updating a component to use container queries</Link>}
+- {<Link href="#full-componentexample" target="_self">Full component example</Link>}
+
+The `setContainerStyles` function helps you generate container query styles based
+on the provided breakpoint and styles. This function is useful for applying
+responsive styles within a container.
+
+## General Information
+
+This guide will walk you through updating a component in Reservoir to replace
+media queries with container queries.
+
+Container queries allow you to apply styles to an element based on the size of
+its container, rather than the size of the viewport. This is useful for creating
+responsive designs that adapt to the size of individual components.
+
+## Updating a component to use container queries
+
+### Defining the container
+
+To define an element as the container add the style prop `container-type: inline-size`
+to the top-most parent.
+
+If the component is using `ComponentWrapper`, add the following prop to the
+definition:
+
+<Source
+  code={`
+sx={{ containerType: "inline-size" }}
+`}
+/>
+
+**Note**: A container cannot change its own styles, so if the top most element has responsive
```

Comment:
Not strictly necessary but try using a `Banner` for notes: https://github.com/NYPL/nypl-design-system/blob/413efcdd1a5fca56c9bd983afd2252f6e4809a15/src/components/SubNav/SubNav.mdx?plain=1#L137